### DISCUSSION TITLE
Having trouble adding labs package to project, any advice?

### DISCUSSION THREAD

**Discussion by kmgallahan** [CommunityToolkit/Labs-Windows#484] [Q&A] [answered]
I'm having trouble using any of the labs packages, not sure what I'm doing wrong here. In particular I'm trying to use StackedNotificationsBehavior. Browsing the labs nuget feed I see this should be under `CommunityToolkit.Labs.WinUI.StackedNotificationsBehavior`, so...

`Directory.Packages.props`:

```
<PackageVersion Include="CommunityToolkit.Labs.WinUI.StackedNotificationsBehavior" Version="0.1.230802" />
```

`Nuget.Config`:

```
<add key="windowscommunitytoolkitlabs" value="https://pkgs.dev.azure.com/dotnet/CommunityToolkit/_packaging/CommunityToolkit-Labs/nuget/v3/index.json" />
...
  <packageSource key="windowscommunitytoolkitlabs">
    <package pattern="CommunityToolkit.Labs.*" />
  </packageSource>
```

`csproj`

```
<PackageReference Include="CommunityToolkit.Labs.WinUI.StackedNotificationsBehavior" />
```

Package sources:

![Screenshot 2023-08-02 160803](https://github.com/CommunityToolkit/Labs-Windows/assets/5379488/5b9f08c4-6632-4e56-b704-5a3b2d4ed27b)

Error message:

```
Error NU1101 Unable to find package CommunityToolkit.WinUI.Behaviors. No packages exist with this id in source(s): nuget.org. PackageSourceMapping is enabled, the following source(s) were not considered: localnugetfeed, sharpgen, windowscommunitytoolkitlabs.
```

If I add `<package pattern="CommunityToolkit.WinUI.Behaviors" />` to the `Nuget.Config` I get this error:

```
Error NU1101 Unable to find package CommunityToolkit.WinUI.Behaviors. No packages exist with this id in source(s): windowscommunitytoolkitlabs. PackageSourceMapping is enabled, the following source(s) were not considered: localnugetfeed, nuget.org, sharpgen.

```

**Comment by kmgallahan** [comment, marked as answer]
I figured it out, you must also add the CommunityToolkit "Latest" feed and map it to use certain "Toolkit Labs" feed packages.

Might be worth adding this to the [Preview Packages](https://github.com/CommunityToolkit/WindowsCommunityToolkit/wiki/Preview-Packages) docs. @michael-hawker

**Reply by michael-hawker** [reply]
Thanks @kmgallahan good point, glad you sorted it out. We just pushed these changes yesterday/today to get everything aligned on the same base code from the new repository as we've been debugging a build issue for the past week+.

Didn't think about the interdependency required for folks consuming packages from the feed, would be nice if NuGet realized this. 🙂

Should get resolved though in the next couple of weeks once we release stable versions for the Windows repo on NuGet as then the Labs packages should point to those instead of the interim builds.

**Reply by kmgallahan** [reply]
@michael-hawker 

Also, last time I built the Toolkit Labs Gallery was in March.

I tried to again just now and ran into a big pile of errors that are probably related to this.

Tried to resolve but gave up, so will be nice once these issues are worked out.
